producer/elasticsearch.go

```diff
@@ -15,297 +15,433 @@
 package producer
 
 import (
-	"bytes"
-	elastigo "github.com/mattbaird/elastigo/lib"
+	"context"
+	"github.com/pkg/errors"
 	"github.com/trivago/gollum/core"
 	"github.com/trivago/tgo"
 	"github.com/trivago/tgo/tcontainer"
-	"strconv"
+	"github.com/trivago/tgo/tlog"
+	"gopkg.in/olivere/elastic.v5"
+	"net/http"
 	"sync"
+	"syscall"
 	"time"
 )
 
 // ElasticSearch producer plugin
 //
 // The ElasticSearch producer sends messages to elastic search using the bulk
-// http API.
+// http API. The producer expects a json payload.
 //
 // Configuration example
 //
-//  - "producer.ElasticSearch":
-//    Connections: 6
-//    RetrySec: 5
-//    TTL: ""
-//    DayBasedIndex: false
+//  producerElasticSearch:
+// 	  Type: producer.ElasticSearch
+//    Retry:
+//		Count: 3
+//		TimeToWaitSec: 3
+//	  SetGzip: true
+//
 //    User: ""
 //    Password: ""
-//    BatchSizeByte: 32768
-//    BatchMaxCount: 256
-//    BatchTimeoutSec: 5
-//    Port: 9200
 //    Servers:
-//      - "localhost"
-//    Index:
-//      "console" : "console"
-//      "_GOLLUM_"  : "_GOLLUM_"
-//    Settings:
-//      "console":
-//        "number_of_shards": 1
-//    DataTypes:
-//      "console":
-//        "source": "ip"
-//    Type:
-//      "console" : "log"
-//      "_GOLLUM_"  : "log"
-//
-// RetrySec denotes the time in seconds after which a failed dataset will be
-// transmitted again. By default this is set to 5.
+//      - http://127.0.0.1:9200
+//    StreamProperties:
+//		<streamName>:
+//			Index: twitter
+// 			DayBasedIndex: false
+//			Type: tweet
 //
-// Connections defines the number of simultaneous connections allowed to a
-// elasticsearch server. This is set to 6 by default.
+// 			# see: https://www.elastic.co/guide/en/elasticsearch/reference/5.4/indices-create-index.html#mappings
+//			Mapping:
+//				# index mapping for payload
+// 				user: keyword
+//				message: text
+//			Settings:
+//				# settings used for mapping
+//				number_of_shards: 5
+//				number_of_replicas: 1
 //
-// TTL defines the TTL set in elasticsearch messages. By default this is set to
-// "" which means no TTL.
+// Retry/Count set the amount of retries before a Elasticsearch request fail finally
 //
-// DayBasedIndex can be set to true to append the date of the message to the
-// index as in "<index>_YYYY-MM-DD". By default this is set to false.
+// Retry/TimeToWaitSec denotes the time in seconds after which a failed dataset will be
+// transmitted again. By default this is set to 3.
 //
-// Servers defines a list of servers to connect to. The first server in the list
-// is used as the server passed to the "Domain" setting. The Domain setting can
-// be overwritten, too.
+// SetGzip enables or disables gzip compression (disabled by default).
```

Comment:
This wording is kind of confusing, at least when read without any prior context. What gets compressed and when?

Maybe we could mention that it corresponds directly to the http://godoc.org/gopkg.in/olivere/elastic.v5#SetGzip SetGzip option in the elastic client library (which enables in-transit compression in the protocol, presumably?)